cirq-google/cirq_google/ops/coupler_pulse.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2020 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""An instance of FSimGate that works naturally on Google's Sycamore chip"""
+
+from typing import Optional, Tuple, TYPE_CHECKING
+
+import numpy as np
+
+from cirq import ops, protocols, value
+
+if TYPE_CHECKING:
+    import cirq
+
+
+_MIN_DURATION = value.Duration(nanos=0)
+_MAX_DURATION = value.Duration(nanos=100)
+
+
+class CouplerPulse(ops.gate_features.TwoQubitGate):
+    """Tunable pulse for entangling adjacent qubits.
+
+    For experimental usage only.
+
+    This operation sends a trapezoidal pulse to the coupler between two
+    adjacent qubits placed in resonance.
+
+    Note that this gate does not have a unitary matrix and must be
+    characterized by the user in order to determine its effects.
+
+    Args:
+            hold_time: Length of the 'plateau' part of the coupler trajectory.
+            coupling_MHz: Target qubit-qubit coupling reached at the plateau.
+            rise_time: Full width of the smoothstep rise/fall.
+            padding_time: Symmetric padding around the coupler pulse.
+    """
+
+    def __init__(
+        self,
+        hold_time: value.Duration,
+        coupling_mhz: float,
+        rise_time: Optional[value.Duration] = value.Duration(nanos=8),
+        padding_time: Optional[value.Duration] = value.Duration(nanos=2.5),
+    ):
```

Comment:
It might be nice to also add the ability to specify single-qubit Z rotations before and after the coupler pulse as part of this gate, since those will generally have to be calibrated in order to use this. Of course, one can always include separate virtual Z rotations in the circuit, but having them be part of the gate might simplify things. Perhaps others with experience using the calibration API can weigh in on this.

Reply:
I am not sure if this is a good idea to do or not.  It is not a convention we follow for any other gates.  We generally specify pre/post Z rotations as separate gates in every other place in the code.

Reply:
My gut feeling is that: "if we calibrate it together, it should be represented by the same concept", thus introducing a phased version of this gate can be useful. 

I.e. the PhasedFSimGate is equivalent to 
```
        0: ───Rz(α0)───FSim(θ, φ)───Rz(β0)───
                       │
        1: ───Rz(α1)───FSim(θ, φ)───Rz(β1)───
```
where FSimGate itself can be looked at as the product of an ISWAPPowGate and a CZPowGate, however during the floquet calibration (eventually) all the 5 parameters are being measured and compensated for, which would be more awkward to do with separate gate classes.

Maybe we could introduce a `PhasedCouplerPulse` that has the two extra parameters to phase pre/post. Maybe that could be introduced after this PR?